src/windows.rs

```diff
@@ -0,0 +1,464 @@
+                device_name: self.device_name.clone(),
+                function_name: "IOCTL_VIDEO_QUERY_SUPPORTED_BRIGHTNESS",
+                last_error: e,
+            }
+            .into()),
+        }
+    }
+}
+
+pub fn brightness_devices() -> impl Stream<Item = Result<Brightness, SysError>> {
+    unsafe extern "system" fn enum_monitors(
+        arg1: HMONITOR,
+        _arg2: HDC,
+        _arg3: *mut RECT,
+        arg4: LPARAM,
+    ) -> BOOL {
+        let monitors = &mut *(arg4.0 as *mut Vec<HMONITOR>);
+        monitors.push(arg1);
+        return true.into();
+    }
+    let mut hmonitors = Vec::<HMONITOR>::new();
+    let devices = unsafe {
+        if !EnumDisplayMonitors(
+            HDC::NULL,
+            std::ptr::null_mut() as *mut RECT,
+            Some(enum_monitors),
+            LPARAM(&mut hmonitors as *mut _ as isize),
+        )
+        .as_bool()
+        {
+            either::Left(iter::once(Err(SysError::ListingDevicesFailed(
+                "EnumDisplayMonitors",
+                GetLastError(),
+            ))))
+        } else {
+            either::Right(hmonitors.into_iter().flat_map(|hmonitor| {
+                // Get the Name of the HMONITOR
+                let mut info: MONITORINFOEXW = std::mem::MaybeUninit::zeroed().assume_init();
+                info.__AnonymousBase_winuser_L13558_C43.cbSize =
+                    std::mem::size_of::<MONITORINFOEXW>() as u32;
+                let info_ptr = &mut info as *mut _;
+                if !GetMonitorInfoW(hmonitor, info_ptr as *mut MONITORINFO).as_bool() {
+                    return vec![Err(SysError::ListingDevicesFailed(
+                        "GetMonitorInfoW",
+                        GetLastError(),
+                    ))];
+                };
+
+                // Get the Physical Monitors in the HMONITOR
+                let mut physical_number: u32 = 0;
+                if !BOOL(GetNumberOfPhysicalMonitorsFromHMONITOR(
+                    hmonitor,
+                    &mut physical_number,
+                ))
+                .as_bool()
+                {
+                    return vec![Err(SysError::ListingDevicesFailed(
+                        "GetNumberOfPhysicalMonitorsFromHMONITOR",
+                        GetLastError(),
+                    ))];
+                };
+                let mut physical_monitors = Vec::with_capacity(physical_number as usize);
+                if !BOOL(GetPhysicalMonitorsFromHMONITOR(
+                    &hmonitor,
+                    physical_number,
+                    physical_monitors.as_mut_ptr(),
+                ))
+                .as_bool()
+                {
+                    return vec![Err(SysError::ListingDevicesFailed(
+                        "GetPhysicalMonitorsFromHMONITOR",
+                        GetLastError(),
+                    ))];
+                }
+                physical_monitors.set_len(physical_number as usize);
+
```

Comment:
From this point on, the code is in charge of a set of handles to physical monitors that need to be closed eventually. If anything panics after this line (and before the code that wraps them in a type implementing `Drop`), the handles are leaked.

Reply:
So now it gets immediately wrapped, so it would be Dropped
https://github.com/stephaneyfx/brightness/blob/2600493ef0bd53d5cd13c84ac7d485b494c27b66/src/windows.rs#L194